### PR TITLE
feat(account invitation): accomplish sign up before accepting the invitation

### PR DESCRIPTION
**Pull request by andreybakanovsky**

### CODE REVIEW

File: app/controllers/accept_account_invitations_controller.rb [hunk 1]
```diff
@@ -16,6 +23,6 @@ def update
   private
 
   helper_method memoize def received_invitation
-    AccountInvitation.unaccepted.for(current_user).find_by!(token: ps.fetch(:token))
+    AccountInvitation.unaccepted.for(current_user).find_by!(token: ps[:token])
```
Comment (ka8725) [inline app/controllers/accept_account_invitations_controller.rb:26]: what was the problem with the previous implementation? revert it?

File: app/controllers/accept_account_invitations_controller.rb [hunk 2]
```diff
@@ -4,8 +4,15 @@ class AcceptAccountInvitationsController < ApplicationController
   def show
     return if user_signed_in?
 
-    session[:after_sign_in_url] = request.fullpath
-    redirect_to new_user_session_url(email: AccountInvitation.find_by!(token: ps.fetch(:token)).email)
+    session[:after_authentication_url] = request.fullpath
+
+    invitee_email = AccountInvitation.find_by!(token: ps[:token]).email
```
Comment (ka8725) [inline app/controllers/accept_account_invitations_controller.rb:9]:
```suggestion
    invitee_email = AccountInvitation.find_by!(token: ps.fetch(:token)).email
```

If the `ps.fetch(:token)` bothers you, create a private method `token_param` or something and use it in all places.

File: app/controllers/accept_account_invitations_controller.rb [hunk 3]
```diff
@@ -4,8 +4,15 @@ class AcceptAccountInvitationsController < ApplicationController
...
+    user = User.find_by(email: invitee_email)
+    if user.present?
+      redirect_to new_user_session_url(email: invitee_email)
+    else
+      redirect_to new_user_registration_url(email: invitee_email)
+    end
```
Comment (ka8725) [inline app/controllers/accept_account_invitations_controller.rb:15]:
```suggestion
    redirect_url = user ? new_user_session_url(email: invitee_email) : new_user_registration_url(email: invitee_email)
    redirect_to redirect_url
```
it's a bit more DRY

**Review by ka8725 — CHANGES_REQUESTED** [review]

File: app/controllers/accept_account_invitations_controller.rb (hunk 1 above)
Comment (andreybakanovsky) [reply app/controllers/accept_account_invitations_controller.rb]: ... ok, I've seen the main idea of the`fetch` for us: errors. Ok

File: app/controllers/accept_account_invitations_controller.rb [hunk 4]
```diff
@@ -4,8 +4,13 @@ class AcceptAccountInvitationsController < ApplicationController
   def show
     return if user_signed_in?
 
-    session[:after_sign_in_url] = request.fullpath
-    redirect_to new_user_session_url(email: AccountInvitation.find_by!(token: ps.fetch(:token)).email)
+    session[:after_authentication_url] = request.fullpath
```
Comment (ka8725) [inline app/controllers/accept_account_invitations_controller.rb:7]: Why changed the key `:after_sign_in_url` to `:after_authentication_url`? `:after_authentication_url` is longer,  harder to write and spell.

**Review by ka8725 — APPROVED** [review]

**Pull request merged**